### DISCUSSION TITLE
Make download models script a subcommand for the binary

### DISCUSSION THREAD

**Discussion by dit7ya** [ggml-org/whisper.cpp#142] [Ideas]
Hi,

Thanks for this awesome project. I am wondering if you could make the bash script part of the binary itself as a subcommand? In that way it will be self-contained and much easier to package for various package-managers.

**Comment by ggerganov** [comment]
We could probably add a `whisper_model_download(const char * name, const char * path)` function to the C-style API.
It can try to download the selected model into the specified path after checking that it is not already there. We probably have to use [std::system](https://en.cppreference.com/w/cpp/utility/program/system) to try to call `curl` or `wget` if available. Not sure what to do if they are not available. Any suggestions? I don't want to add a third-party dependency to the library, just to be able to download the model files.

**Reply by dit7ya** [reply]
Thanks. I think it is fine to depend on curl or wget, and show a failed message if they are not available.
